components/search_results/search_results.tsx

```diff
@@ -0,0 +1,297 @@
+                    {'sidebar-expanded': isSideBarExpanded},
+                ])}
+            >
+                <NoResultsIndicator {...noResultsprops}/>
+            </div>
+        );
+        break;
+    default:
+        ctls = sortedResults.map((post, index) => {
+            return (
+                <SearchResultsItem
+                    key={post.id}
+                    compactDisplay={props.compactDisplay}
+                    post={post}
+                    matches={props.matches[post.id]}
+                    term={(!props.isFlaggedPosts && !props.isPinnedPosts && !props.isMentionSearch) ? searchTerms : ''}
+                    isMentionSearch={props.isMentionSearch}
+                    a11yIndex={index}
+                    isFlaggedPosts={props.isFlaggedPosts}
+                    isPinnedPosts={props.isPinnedPosts}
+                />
+            );
+        });
+
+        loadingMorePostsComponent = (showLoadMore) ? (
+            <div className='loading-screen'>
+                <div className='loading__content'>
+                    <div className='round round-1'/>
+                    <div className='round round-2'/>
+                    <div className='round round-3'/>
+                </div>
+            </div>
+        ) : null;
+    }
+
+    return (
+        <div
+            id='searchContainer'
+            className='sidebar-right__body'
+        >
+            <SearchResultsHeader>
+                {formattedTitle}
+                {props.channelDisplayName && <div className='sidebar--right__title__channel'>{props.channelDisplayName}</div>}
+            </SearchResultsHeader>
+            <Scrollbars
+                ref={scrollbars}
+                autoHide={true}
+                autoHideTimeout={500}
+                autoHideDuration={500}
+                renderThumbHorizontal={renderThumbHorizontal}
+                renderThumbVertical={renderThumbVertical}
+                renderView={renderView}
+                onScroll={handleScroll}
+            >
+                <div
+                    id='search-items-container'
+                    role='application'
+                    className={classNames(['search-items-container post-list__table a11y__region', {'no-results': noResults}])}
+                    data-a11y-sort-order='3'
+                    data-a11y-focus-child={true}
+                    data-a11y-loop-navigation={false}
+                    aria-label={props.intl.formatMessage({
+                        id: 'accessibility.sections.rhs',
+                        defaultMessage: '{regionTitle} complimentary region',
+                    }, {
+                        regionTitle: formattedTitle,
+                    })}
+                >
+                    {ctls}
+                    {loadingMorePostsComponent}
+                </div>
+            </Scrollbars>
+        </div>
+    );
+};
+
+React.memo(SearchResults, shouldRenderFromProps);
```

Comment:
👏

Reply:
@michelengelen, I think the return of `React.memo` is what needs to be used here. E.g.:
```tsx
export default React.memo(SearchResults, shouldRenderFromProps);
```

Reply:
you are absolutely right ... nice catch